src/api/spec/controllers/person_controller_spec.rb

```diff
@@ -5,9 +5,27 @@
 # CONFIG['global_write_through'] = true
 
 RSpec.describe PersonController, vcr: false do
-  describe 'POST #post_userinfo' do
-    let(:user) { create(:confirmed_user) }
+  let(:user) { create(:confirmed_user) }
+  let(:admin_user) { create(:admin_user) }
+
+  let!(:old_realname) { user.realname }
+  let!(:old_email) { user.email }
+
+  shared_examples "not allowed to change user details" do
+    it 'sets an error code' do
+      expect(response.header['X-Opensuse-Errorcode']).to eq('change_userinfo_no_permission')
+    end
 
+    it 'does not change users real name' do
+      expect(user.realname).to eq(old_realname)
```

Comment:
Do not re-invent the rspec [change matcher](https://relishapp.com/rspec/rspec-expectations/v/3-6/docs/built-in-matchers/change-matcher)